core/src/main/java/org/apache/iceberg/ContentFileParser.java

```diff
@@ -27,7 +27,7 @@
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.util.JsonUtil;
 
-class ContentFileParser {
+public class ContentFileParser {
```

Comment:
I think it's OK to make this public so we can reuse it and have the new parsers in the REST module.

Reply:
The alternative is to to move the unbound methods into the TableScanResponseParser. Can revisit this later if needed but for now i'd say let's go with it for this specific PR, I may change to this in the next one

Reply:
i take my alternative back, we wouldn't be able to use GenericDataFile/GenericDeleteFile in the rest module